src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -317,6 +328,20 @@ object HashMap extends ImmutableMapFactory[HashMap] with BitOperations.Int {
         None
     }
 
+    override protected def contains0(key: A, hash: Int, level: Int): Boolean = {
+      // Note: this code is duplicated with `get0`
+      val index = (hash >>> level) & 0x1f
+      val mask = (1 << index)
+      if (bitmap == - 1) {
+        elems(index & 0x1f).contains0(key, hash, level + 5)
+      } else if ((bitmap & mask) != 0) {
+        val offset = Integer.bitCount(bitmap & (mask-1))
+        elems(offset).contains0(key, hash, level + 5)
```

Comment:
@Ichoran, I fully agree, even if it would be faster, it made Vector very obfuscated, in the end you guys will have to decide whether it's worth the trouble.
@retronym, would you like me to try this out, or would you like to?